src/main/java/com/reservation/hall/domain/Hall.java

```diff
@@ -0,0 +1,72 @@
+package com.reservation.hall.domain;
+
+import com.reservation.hall.domain.exception.InvalidHallStatusChangeException;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import java.util.Objects;
+import java.util.Set;
+
+@Getter
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+public class Hall {
+    @EqualsAndHashCode.Include
+    private final HallId id;
+    private HallName name;
+    private Capacity capacity;
+    private Set<Equipment> equipment ;
+    private HallStatus status;
+
+    private Hall(HallId id, HallName name, Capacity capacity, Set<Equipment> equipment) {
+        this.id = Objects.requireNonNull(id);
+        this.name = Objects.requireNonNull(name);
+        this.capacity = capacity;
+        this.equipment = Set.copyOf(Objects.requireNonNull(equipment));
+        this.status = HallStatus.ACTIVE;
+    }
+
+    public static Hall create(HallId id, HallName name, Capacity capacity, Set<Equipment> equipment) {
+        return new Hall(id, name, capacity, equipment);
+    }
+
+    public void rename(HallName name) {
+        this.name = Objects.requireNonNull(name);
+    }
+
+    public void resize(Capacity capacity) {
+        this.capacity = capacity;
+    }
```

Comment:
Brakujący null check
```suggestion
    public void resize(Capacity capacity) {
        this.capacity = Objects.requireNonNull(capacity);
    }
```